core/src/test/java/org/apache/calcite/util/SourceTest.java

```diff
@@ -67,6 +70,19 @@ private static String getRootPrefix() {
     }
   }
 
+  @Test public void testFileToUrl() throws URISyntaxException {
+    final String path = "abc.txt";
+    Source fileSource = file(null, path);
+    URL url = fileSource.url();
+
+    assertFalse(url == null, "No url generated for file source");
+
+    // convert the url back to a file, and assert they are identical.
+    File urlFile = new File(url.toURI().getPath());
```

Comment:
Does this support filenames with spaces?

Reply:
Yes. The standard Java API supports file names with spaces.
I have revised the test case to give a file name with space in it. 
Please take a look. Thank you.